pkg/handlers/s3_handler.go

```diff
@@ -103,6 +104,11 @@ func (h *s3Handler) OnAdd(ns string, vc kvcv1.VolumeConfig, controllerRef metav1
 		recursiveFlag = "--recursive"
 	}
 
+	s3Regexp := regexp.MustCompile("s3://(\\w+)/?(.*)")
+	bucketArray := s3Regexp.FindAllStringSubmatch(vc.SourceURL, -1)
+	bucketName := bucketArray[0][1]
+	bucketPath := bucketArray[0][2]
```

Comment:
Can we error check and error out if the s3 source is not valid?

Reply:
Or maybe we can use https://github.com/minio/minio-go or some other library to parse the s3 url and get the bucket name and path.

Reply:
That sounds like a feature request, I can do it in another pull

Reply:
Ok, we should error check it before getting it in though.

Reply:
you were right, fixed.